web/client/plugins/__tests__/DynamicLegend-test.jsx

```diff
@@ -0,0 +1,541 @@
+            const { Plugin } = getPluginForTest(DynamicLegend, { ...SELECTED_LAYER_STATE, controls: { widgetBuilder: {available: true}}});
+            const WrappedPlugin = dndContext(Plugin);
+            const items = getTOCItems({ WidgetsBuilder });
+            ReactDOM.render(<WrappedPlugin items={items} />, document.getElementById("container"));
+            // check tools
+
+            expect(document.querySelector(ZOOM_TO_SELECTOR)).toBeTruthy("zoom doesn't exist");
+            expect(document.querySelector(WIDGET_BUILDER_SELECTOR)).toBeTruthy("widget doesn't exist");
+            expect(document.querySelector(REMOVE_SELECTOR)).toBeTruthy("remove doesn't exist");
+        });
+        it('should render the zoom to layer button if there is a vector layer with valid features', () => {
+            const { Plugin } = getPluginForTest(DynamicLegend, {
+                layers: {
+                    flat: [
+                        {
+                            id: 'topp:states__6',
+                            name: 'topp:states',
+                            type: 'vector',
+                            visibility: true,
+                            features: [{
+                                type: 'Feature',
+                                properties: {},
+                                geometry: {
+                                    type: 'Point',
+                                    coordinates: [0, 0]
+                                }
+                            }]
+                        }
+                    ],
+                    groups: [
+                        {
+                            id: 'Default',
+                            title: 'Default',
+                            name: 'Default',
+                            nodes: [
+                                'topp:states__6'
+                            ],
+                            expanded: true
+                        }
+                    ],
+                    selected: [
+                        'topp:states__6'
+                    ],
+                    settings: {
+                        expanded: false,
+                        node: null,
+                        nodeType: null,
+                        options: {}
+                    },
+                    layerMetadata: {
+                        expanded: false,
+                        metadataRecord: {},
+                        maskLoading: false
+                    }
+                }
+            });
+            const WrappedPlugin = dndContext(Plugin);
+            ReactDOM.render(<WrappedPlugin />, document.getElementById("container"));
+            // check zoom and remove selector
+            expect(document.querySelectorAll(TOOL_BUTTON_SELECTOR).length).toBe(3);
+            expect(document.querySelector(ZOOM_TO_SELECTOR)).toBeTruthy();
+            expect(document.querySelector(REMOVE_SELECTOR)).toBeTruthy();
+        });
+    });
+});
```

Comment:
testing locally most of these tests are failing. These tests seems a copy of the [TOC-test.jsx](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/plugins/__tests__/TOC-test.jsx) files. Even if the DynamicLegend is using the TOC as main component we cannot replicate the same unit tests of the TOC plugin because they have different behavior and structure (e.g. most of these test are verifying correct injection of tools inside the toolbar not present in the DynamicLegend plugin).
These tests needs to be removed and replaced with tests specific for DynamicLegend plugin